src/refdb_fs.c

```diff
@@ -1090,22 +1090,27 @@ static int packed_write(refdb_fs_backend *backend)
 static int packed_delete(refdb_fs_backend *backend, const char *ref_name)
 {
 	size_t pack_pos;
-	int error;
+	int error, found = 0;
 
 	if ((error = packed_reload(backend)) < 0)
 		goto cleanup;
 
-	/* If a packed reference exists, remove it from the packfile and repack */
 	if ((error = git_sortedcache_wlock(backend->refcache)) < 0)
 		goto cleanup;
 
-	if (!(error = git_sortedcache_lookup_index(
-			&pack_pos, backend->refcache, ref_name)))
+	/* If a packed reference exists, remove it from the packfile and repack if necessary */
+	error = git_sortedcache_lookup_index(&pack_pos, backend->refcache, ref_name);
+	if (error == 0) {
 		error = git_sortedcache_remove(backend->refcache, pack_pos);
+		found = 1;
+	}
+	if (error == GIT_ENOTFOUND)
+		error = 0;
```

Comment:
Wouldn't the general flow be a bit easier if we'd not introduce a `found` but a `locked` variable, instead?

```
static int packed_delete(refdb_fs_backend *backend, const char *ref_name)
{
	int error, locked = 0;
	size_t pack_pos;

	if ((error = packed_reload(backend)) < 0)
		goto cleanup;

	/* If a packed reference exists, remove it from the packfile and repack */
	if ((error = git_sortedcache_wlock(backend->refcache)) < 0)
		goto cleanup;

	locked = 1;

	if ((error = git_sortedcache_lookup_index(&pack_pos, backend->refcache, ref_name)) < 0) {
		if (error == GIT_ENOTFOUND)
			error = 0;
		goto cleanup;
	}

	if ((error = git_sortedcache_remove(backend->refcache, pack_pos)) < 0)
		goto cleanup;

	if ((error = packed_write(backend)) < 0)
		goto cleanup;

cleanup:
	if (locked)
		git_sortedcache_wunlock(backend->refcache);
	return error;
}
```

Might just be me, though